errdefs/is.go

```diff
@@ -28,6 +28,32 @@ func getImplementer(err error) error {
 	}
 }
 
+// IsHTTPOnlyError checks whether err is one of the un-exported HTTP error type provided by this package. These errors
+// only define what HTTP status code should be returned, but don't add additional context to the actual error they
+// wrap. By contrast, error types implementing the interfaces defined in this package are also adding such additional
+// context.
+func IsHTTPOnlyError(err error) bool {
```

Comment:
Also wondering what would make these errors different from, say, https://github.com/moby/moby/blob/2b449e0e6512c70ac783450ffb891d940c29de51/builder/builder-next/builder.go#L42-L46

Which wouldn't be caught by this function. The non-exported errors in this package are mostly for convenience (and "one" implementation of errdefs types), but there's many others, and the information they add is to put an error in a specific category (see my other comment above).

Reply:
These errors all have the property that `errors.Unwrap(err).Error() == err.Error()`, implement one of the `errdefs` interfaces (which map 1:1 to HTTP status codes, except for HTTP 500), _and nothing else_. `errMultipleFilterValues` is not caught by this function _by design_, **because** it carries additional context. The information carried by the non-exported convenience types is already encoded in the response, **in the HTTP status code.** Duplicating that in the error tree is therefore just redundant noise.

Reply:
>  The information carried by the non-exported convenience types is already encoded in the response, in the HTTP status code. Duplicating that in the error tree is therefore just redundant noise.


But that only applies to the final error, and not intermediate ones, and currently we're discarding that information; see for example this test-case;

```go
"error wrapped multiple times with a HTTP-only error": {
	err:      errdefs.InvalidParameter(errdefs.Conflict(errors.New("foobar"))),
	expected: &types.ErrorResponse{Message: "foobar"},
},
```